pandas/core/arrays/datetimelike.py

```diff
@@ -1495,7 +1495,17 @@ def __rsub__(self, other):
             return (-self) + other
 
         # We get here with e.g. datetime objects
-        return -(self - other)
+        from pandas.core.arrays import DatetimeArray
+
+        datetime_result = self - other
+        try:
+            return -(datetime_result)
+        except TypeError as e:
+            if isinstance(datetime_result, DatetimeArray):
```

Comment:
I don't think you need this `if` statement here. We'd always want to raise the exception